### ISSUE TITLE
[Accessibility], WCAG 1.3.2 and 2.4.3. Search across vocabularies -expandable button’s contents are not logically in focus order

### ISSUE THREAD

**Issue by Eiskore** [NatLibFi/Skosmos#1970]
## URL address of the page where you encountered the problem
https://test.dev.finto.fi/en/ 

## Description of the problem
In the Navigation, there is an expandable button " Search across the vocabularies /Yhteishaku sanastoista". The contents of the expanded button are not logically next in the focus order, but the other links of the navigation are first focused and read with a screen reader before moving  to the expanded content.

Fix: The contents of the expanded button should be focused immediately after the expansion button. Otherwise, especially with a screen reader, the reading order is illogical, and the screen reader may cause uncertainty as to whether the expandable button has been activated at all if the contents of the expanded button are not immediately found in connection with the triggering button.  

How to test: Use only the Tab key to move around. When the Yhteishaku sanastoista / Search across vocabularies -button is focused, press space and after that press tab key:

Expected result: Keyboard focus moves to “Choose vocabulary menu”.

Actual result: Keyboard focus moves to Vocabularies -link.

Severity: moderate
Priority: medium

## Additional information (e.g. screenshots) about the problem

<img width="1004" height="122" alt="Image" src="https://github.com/user-attachments/assets/eecc4450-b950-46ce-9ed3-c9609789dc37" />

## The browser you used when the problem appeared
Chrome, Firefox, Edge

**Comment by tinskuu** [comment]
The issue is not fixed.
